src/vs/workbench/contrib/chat/browser/chatSessions/view/sessionsViewPane.ts

```diff
@@ -70,6 +73,204 @@ class SessionsAccessibilityProvider {
 	}
 }
 
+// Word-based fuzzy find provider for sessions
+class ChatSessionsFindProvider implements IAsyncFindProvider<ChatSessionItemWithProvider> {
+	private pattern: string = '';
+	private matchType: TreeFindMatchType = TreeFindMatchType.Contiguous;
+	private matchedSessions: Set<ChatSessionItemWithProvider> = new Set();
+	private tree: WorkbenchAsyncDataTree<IChatSessionItemProvider, ChatSessionItemWithProvider, FuzzyScore> | undefined;
+	private isSearchActive: boolean = false;
+
+	constructor(
+		private readonly provider: IChatSessionItemProvider,
+	) { }
+
+	setTree(tree: WorkbenchAsyncDataTree<IChatSessionItemProvider, ChatSessionItemWithProvider, FuzzyScore>): void {
+		this.tree = tree;
+	}
+
+	/**
+	 * Converts a session to searchable text combining label, id and description
+	 */
+	private getSearchableText(session: ChatSessionItemWithProvider): string {
+		const parts = [
+			session.label || '',
+			session.id || '',
+			typeof session.description === 'string' ? session.description : (session.description?.value || '')
+		];
+		return parts.filter(text => text.length > 0).join(' ');
+	}
+
+	/**
+	 * Word-based fuzzy matching: each search word must appear in the searchable text in order
+	 * but can be separated by other text
+	 */
+	private matches(pattern: string, searchableText: string, matchType: TreeFindMatchType): boolean {
+		if (!pattern) {
+			return true;
+		}
+
+		const textLower = searchableText.toLowerCase();
+
+		if (matchType === TreeFindMatchType.Contiguous) {
+			// Exact substring match
+			return textLower.includes(pattern.toLowerCase());
+		}
+
+		// Word-based fuzzy matching (default)
+		const words = pattern.split(/\s+/).filter(w => w.length > 0);
+		let currentPosition = 0;
+
+		// Each word must appear in order in the text
+		for (const word of words) {
+			const wordLower = word.toLowerCase();
+			const foundIndex = textLower.indexOf(wordLower, currentPosition);
+
+			if (foundIndex === -1) {
+				return false; // Word not found
+			}
+
+			currentPosition = foundIndex + wordLower.length;
+		}
+
+		return true; // All words found in order
+	}
+
+	/**
+	 * Gets all rendered sessions from the tree's visible nodes, including nested/collapsed items
+	 */
+	private getAllSessionsFromTree(): ChatSessionItemWithProvider[] {
+		if (!this.tree) {
+			return [];
+		}
+
+		const sessions: ChatSessionItemWithProvider[] = [];
+
+		// Get the provider node
+		if (!this.tree.hasNode(this.provider)) {
+			return sessions;
+		}
+
+		// Recursively collect all session nodes
+		const collectSessions = (node: ITreeNode<ChatSessionItemWithProvider | IChatSessionItemProvider, FuzzyScore>) => {
```

Comment:
we are using a tree here since it's a simple `Provider -> Sessions[]` relationship, we can loop through the children of `providerNode` without the recursion.